torch_xla/csrc/ir.h

```diff
@@ -151,11 +151,12 @@ class XlaNode : public torch::lazy::Node {
   xla::Shape xla_shape_;
   torch::lazy::hash_t node_hash_ = 0;
   torch::lazy::hash_t dag_hash_;
-  torch::lazy::hash_t sharding_hash_ = 0;
 
   // Experimental sharding annotation attached to the IR node.
   // TODO(yeounoh): make sure that view update doesn't reset this.
   std::shared_ptr<xla::OpSharding> output_sharding_ = nullptr;
+
+  std::optional<torch::lazy::OpList> oplist_;
```

Comment:
I guess a major issue we need to work around would be keeping the cross references between ops up-to-date, given that the graph is a tree. It might be doable with additional bookkeeping and traversals, but not sure how expensive that would be.

Reply:
I think the best way to do it will be storing ops as smart pointers, either shared or weak, so that we can access them within an oplist by index while also being able to replace the node with a new node.